databricks/koalas/namespace.py

```diff
@@ -2499,6 +2501,275 @@ def broadcast(obj):
+    # +-----------------------------------+-------------------+-------------------+-------------...
+
+    # In above `sdf`, the type of `__index_merged_____columns_merged__` is `string`, not `struct`
+    # Therefore, we need to make `struct` column for addressing MultiIndex.
+    merged_col = index_name + "_" + columns_name
+
+    sdf_struct_map = sdf.select(
+        sdf[index_name], F.regexp_replace(sdf[index_name].cast("string"), " ", "").alias(merged_col)
+    ).dropDuplicates()
+    # `sdf_struct_map` here is shown like below.
+    # +----------------+-----------------------------------+
+    # |__index_merged__|__index_merged_____columns_merged__|
+    # +----------------+-----------------------------------+
+    # | [bar, bar, bar]|                      [bar,bar,bar]|
+    # | [foo, foo, foo]|                      [foo,foo,foo]|
+    # +----------------+-----------------------------------+
+    data_columns = sdf_crosstab.columns[1:]
+
+    sdf = sdf_crosstab.join(
+        sdf_struct_map, sdf_crosstab[merged_col] == sdf_struct_map[merged_col]
+    ).select(index_name, *data_columns)
+    # now, `sdf` here is shown like below which has a struct column `__index_merged__`
+    # +----------------+-------------------+-------------------+---------------------+
+    # |__index_merged__|[dull,one,one,dull]|[dull,two,two,dull]|[shiny,two,two,shiny]|
+    # +----------------+-------------------+-------------------+---------------------+
+    # | [bar, bar, bar]|                  1|                  1|                    0|
+    # | [foo, foo, foo]|                  2|                  1|                    2|
+    # +----------------+-------------------+-------------------+---------------------+
+
+    # but if given `index` as parameter is single value (not list-like),
+    # `sdf` here is shown like below.
+    # +-----+----+-----+
+    # |row_0|dull|shiny|
+    # +-----+----+-----+
+    # |  foo|   3|    4|
+    # |  bar|   2|    2|
+    # +-----+----+-----+
+
+    if len(tmp_index_names) > 1:
+        sdf = sdf.select(F.expr("__index_merged__.*"), *data_columns)
+    # fianlly, `sdf` here is shown like below
+    # +-----+-----+-----+-------------------+-------------------+---------------------+
+    # |row_0|row_1|row_2|[dull,one,one,dull]|[dull,two,two,dull]|[shiny,two,two,shiny]|
+    # +-----+-----+-----+-------------------+-------------------+---------------------+
+    # |  bar|  bar|  bar|                  1|                  1|                    0|
+    # |  foo|  foo|  foo|                  2|                  1|                    2|
+    # +-----+-----+-----+-------------------+-------------------+---------------------+
+
+    internal = InternalFrame(
+        spark_frame=sdf,
+        index_map=OrderedDict((index_name, (index_name,)) for index_name in tmp_index_names),
+        column_labels=[
+            tuple(col.replace("[", "").replace("]", "").split(",")) for col in data_columns
```

Comment:
This line is so hacky but hard to resolve....😂

Let's say we have `data_columns` like below.

```python
>>> data_columns
['[one,bar]', '[one,foo]', '[two,bar]', '[two,foo]']
```

we have to transform this to "column_labels_like" format like below.

```python
>>> column_labels_like
[('one', 'bar'), ('one', 'foo'), ('two', 'bar'), ('two', 'foo')]
```

Could someone give me an any idea??